core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -104,6 +104,9 @@ public boolean equals(Object o) {
       return true;
     }
     // allow equality with any Set.
+    if (!(o instanceof ConcurrentHashSet)) {
```

Comment:
Actually since we extend `AbstractSet`, we shouldn't even need to implement `equals` or `hashCode`.

https://docs.oracle.com/javase/7/docs/api/java/util/AbstractSet.html#equals(java.lang.Object)